lib/internal/http2/core.js

```diff
@@ -3095,7 +3099,7 @@ Http2Server.prototype[EventEmitter.captureRejectionSymbol] = function(
     case 'stream':
       // TODO(mcollina): we might want to match this with what we do on
       // the compat side.
-      const [stream] = args;
+      const { 0: stream } = args;
```

Comment:
Isn't this also "vulnerable" to prototype pollution "attacks," if we really want to consider that "unsafe?" I mean, one could define a getter on the `Array.prototype` that returns something "malicious."

I'm really not sure if all this complexity and illegibility are worth the minimal benefit.

Reply:
Defining a `0` property on the `Array.prototype` would have effect only on empty arrays, right? `args` here is created by the spread operator, so I think this code would actually be safe even in case of `Array.prototype` pollution.


```js
Object.defineProperty(Array.prototype, '0', {get(){return 9}, set(v){/* no op */}})
((t, ...args)=>args[0])(1, 2, 3) === 2 // true
```

I don't think we are concerned about code being "malicious" anyway, what we are trying to achieve with the move to primordials is to allow users to monkey-patch the built-in objects (for debugging purposes, for tinkering, etc.) without having errors thrown by Node.js internals.

Reply:
> so I think this code would actually be safe even in case of `Array.prototype` pollution.

Might be, I am not sure.

> without having errors thrown by Node.js internals.

I understand the motivation, but I am not sure it's worth the cost to maintainability, readability etc.

Reply:
To address the maintainability concern, would you prefer if we added a linter rule to forbid array destructuring assignment?